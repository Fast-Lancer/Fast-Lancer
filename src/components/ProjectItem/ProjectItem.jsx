import styles from './ProjectItem.module.css'

export default function ProjectItem({ project }) {
  const { title, clients, hourly_rate, hours_quoted, price_quoted } = project

  return (
    <div className={styles.projectItem}>
      <div className={styles.col}>
        <h3 className={styles.h3}>
          {title}
        </h3>
        <h4 className={styles.h4}>
          {clients.client_name}
        </h4>
      </div>
      <div className={styles.col}>
        <div>Hourly Rate: {hourly_rate}</div>
        <div>Hours Quoted: {hours_quoted}</div>
        <div>Quoted Price: {price_quoted}</div>
      </div>
    </div>
  )
}
