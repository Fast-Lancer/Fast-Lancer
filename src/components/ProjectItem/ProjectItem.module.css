.projectItem {
  display: grid;
  grid-template-columns: 2fr 1fr;
  align-items: center;
  border-bottom: 1px black solid;
  padding: 2vmax;
}

.titleSection {
  position: relative;
  left: 1vw;
}

.projectItem > section > * {
  padding: 0.1vh;
  margin: 0;
}

.clientName {
  text-indent: 1vw;
  color: darkgray;
}