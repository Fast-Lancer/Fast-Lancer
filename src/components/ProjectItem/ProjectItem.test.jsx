import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import ProjectItem from './ProjectItem.jsx'


it('should display a project item with provided content', () => {
  const { container } = render(
    <MemoryRouter>
      <ProjectItem project={{ title: 'something', clients: { name: 'person' }, hourly_rate: 33, hours_quoted: 55, price_quoted: 375 }}/>
    </MemoryRouter>
  )

  expect(container).toMatchSnapshot()
})
