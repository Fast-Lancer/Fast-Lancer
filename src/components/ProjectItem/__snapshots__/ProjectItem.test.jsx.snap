// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a project item with provided content 1`] = `
<div>
  <div
    class="projectItem"
  >
    <div
      class="col"
    >
      <h3
        class="h3"
      >
        something
      </h3>
      <h4
        class="h4"
      />
    </div>
    <div
      class="col"
    >
      <div>
        Hourly Rate: 
        33
      </div>
      <div>
        Hours Quoted: 
        55
      </div>
      <div>
        Quoted Price: 
        375
      </div>
    </div>
  </div>
</div>
`;
